### YamlMime:ManagedReference
items:
- uid: System.Security.Cryptography.CryptographicException
  id: CryptographicException
  children:
  - System.Security.Cryptography.CryptographicException.#ctor
  - System.Security.Cryptography.CryptographicException.#ctor(System.Int32)
  - System.Security.Cryptography.CryptographicException.#ctor(System.String)
  - System.Security.Cryptography.CryptographicException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  - System.Security.Cryptography.CryptographicException.#ctor(System.String,System.Exception)
  - System.Security.Cryptography.CryptographicException.#ctor(System.String,System.String)
  langs:
  - csharp
  name: CryptographicException
  nameWithType: CryptographicException
  fullName: System.Security.Cryptography.CryptographicException
  type: Class
  summary: "Исключение, которое выдается при возникновении ошибки во время криптографической операции."
  remarks: "CryptographicException использует HRESULT CORSEC_E_CRYPTO, только если указан код альтернативного ошибки.       Список начальных значений свойств для экземпляра CryptographicException см."
  example:
  - "The following code example demonstrates how to use members of the CryptographicException class.  \n  \n [!code-cs[System.Security.Cryptography.CryptographicException#2](~/add/codesnippet/csharp/t-system.security.crypto_115_1.cs)]\n [!code-vb[System.Security.Cryptography.CryptographicException#2](~/add/codesnippet/visualbasic/t-system.security.crypto_115_1.vb)]\n [!code-cpp[System.Security.Cryptography.CryptographicException#2](~/add/codesnippet/cpp/t-system.security.crypto_115_1.cpp)]"
  syntax:
    content: >-
      [System.Runtime.InteropServices.ComVisible(true)]

      public class CryptographicException : SystemException
  inheritance:
  - System.Exception
  - System.SystemException
  implements: []
  inheritedMembers: []
  platform:
  - net462
- uid: System.Security.Cryptography.CryptographicException.#ctor
  id: '#ctor'
  parent: System.Security.Cryptography.CryptographicException
  langs:
  - csharp
  name: CryptographicException()
  nameWithType: CryptographicException.CryptographicException()
  fullName: System.Security.Cryptography.CryptographicException.CryptographicException()
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Security.Cryptography
  summary: "Инициализирует новый экземпляр <xref href=&quot;System.Security.Cryptography.CryptographicException&quot;> </xref> класса со свойствами по умолчанию."
  remarks: "В следующей таблице показаны исходные значения свойств для экземпляра класса <xref:System.Security.Cryptography.CryptographicException>.</xref:System.Security.Cryptography.CryptographicException>      | Свойство | Значение |   |--------------|-----------|   | <xref:System.Exception.InnerException%2A>| `null`.|   | <xref:System.Exception.Message%2A>| Строка локализованного сообщения об. |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A>"
  example:
  - "The following code example demonstrates how to construct a <xref:System.Security.Cryptography.CryptographicException> with no parameters. This code example is part of a larger example provided for the <xref:System.Security.Cryptography.CryptographicException> class.  \n  \n [!code-cs[System.Security.Cryptography.CryptographicException#1](~/add/codesnippet/csharp/m-system.security.crypto_47_1.cs)]\n [!code-vb[System.Security.Cryptography.CryptographicException#1](~/add/codesnippet/visualbasic/m-system.security.crypto_47_1.vb)]\n [!code-cpp[System.Security.Cryptography.CryptographicException#1](~/add/codesnippet/cpp/m-system.security.crypto_47_1.cpp)]"
  syntax:
    content: public CryptographicException ();
    parameters: []
  overload: System.Security.Cryptography.CryptographicException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Security.Cryptography.CryptographicException.#ctor(System.Int32)
  id: '#ctor(System.Int32)'
  parent: System.Security.Cryptography.CryptographicException
  langs:
  - csharp
  name: CryptographicException(Int32)
  nameWithType: CryptographicException.CryptographicException(Int32)
  fullName: System.Security.Cryptography.CryptographicException.CryptographicException(Int32)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Security.Cryptography
  summary: "Инициализирует новый экземпляр <xref href=&quot;System.Security.Cryptography.CryptographicException&quot;> </xref> класса с заданным <xref uid=&quot;langword_csharp_HRESULT&quot; name=&quot;HRESULT&quot; href=&quot;&quot;> </xref> код ошибки."
  remarks: "Конструктор CryptographicException принимает системы `HRESULT` код ошибки и устанавливает <xref:System.Exception.Message%2A>локализованное сообщение, которое соответствует свойству `HRESULT`.</xref:System.Exception.Message%2A>       В следующей таблице показаны исходные значения свойств для экземпляра класса <xref:System.Security.Cryptography.CryptographicException>.</xref:System.Security.Cryptography.CryptographicException>      | Свойство | Значение |   |--------------|-----------|   | <xref:System.Exception.InnerException%2A>| `null`.|   | <xref:System.Exception.Message%2A>| Строка локализованного сообщения об. |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A>"
  example:
  - "The following code example demonstrates how to construct a <xref:System.Security.Cryptography.CryptographicException> using an error code for an <xref:System.Security.Cryptography.CryptographicUnexpectedOperationException>. This code example is part of a larger example provided for the <xref:System.Security.Cryptography.CryptographicException> class.  \n  \n [!code-cs[System.Security.Cryptography.CryptographicException#3](~/add/codesnippet/csharp/m-system.security.crypto_36_1.cs)]\n [!code-vb[System.Security.Cryptography.CryptographicException#3](~/add/codesnippet/visualbasic/m-system.security.crypto_36_1.vb)]\n [!code-cpp[System.Security.Cryptography.CryptographicException#3](~/add/codesnippet/cpp/m-system.security.crypto_36_1.cpp)]"
  syntax:
    content: public CryptographicException (int hr);
    parameters:
    - id: hr
      type: System.Int32
      description: "<xref uid=&quot;langword_csharp_HRESULT&quot; name=&quot;HRESULT&quot; href=&quot;&quot;> </xref> Код ошибки."
  overload: System.Security.Cryptography.CryptographicException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Security.Cryptography.CryptographicException.#ctor(System.String)
  id: '#ctor(System.String)'
  parent: System.Security.Cryptography.CryptographicException
  langs:
  - csharp
  name: CryptographicException(String)
  nameWithType: CryptographicException.CryptographicException(String)
  fullName: System.Security.Cryptography.CryptographicException.CryptographicException(String)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Security.Cryptography
  summary: "Инициализирует новый экземпляр <xref href=&quot;System.Security.Cryptography.CryptographicException&quot;> </xref> класса с указанным сообщением об ошибке."
  remarks: "В следующей таблице показаны исходные значения свойств для экземпляра класса <xref:System.Security.Cryptography.CryptographicException>.</xref:System.Security.Cryptography.CryptographicException>      | Свойство | Значение |   |--------------|-----------|   | <xref:System.Exception.InnerException%2A>| `null`.|   | <xref:System.Exception.Message%2A>| Строка сообщения об ошибке. |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A>"
  example:
  - "The following code example demonstrates how to construct a <xref:System.Security.Cryptography.CryptographicException> using a custom error message. This code example is part of a larger example provided for the <xref:System.Security.Cryptography.CryptographicException> class.  \n  \n [!code-cs[System.Security.Cryptography.CryptographicException#4](~/add/codesnippet/csharp/m-system.security.crypto_116_1.cs)]\n [!code-vb[System.Security.Cryptography.CryptographicException#4](~/add/codesnippet/visualbasic/m-system.security.crypto_116_1.vb)]\n [!code-cpp[System.Security.Cryptography.CryptographicException#4](~/add/codesnippet/cpp/m-system.security.crypto_116_1.cpp)]"
  syntax:
    content: public CryptographicException (string message);
    parameters:
    - id: message
      type: System.String
      description: "Сообщение об ошибке с объяснением причин исключения."
  overload: System.Security.Cryptography.CryptographicException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Security.Cryptography.CryptographicException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  id: '#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)'
  parent: System.Security.Cryptography.CryptographicException
  langs:
  - csharp
  name: CryptographicException(SerializationInfo,StreamingContext)
  nameWithType: CryptographicException.CryptographicException(SerializationInfo,StreamingContext)
  fullName: System.Security.Cryptography.CryptographicException.CryptographicException(SerializationInfo,StreamingContext)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Security.Cryptography
  summary: "Инициализирует новый экземпляр <xref href=&quot;System.Security.Cryptography.CryptographicException&quot;> </xref> сериализованными данными."
  remarks: "Данный конструктор вызывается в процессе десериализации для воссоздания объекта исключения, переданного в потоке."
  syntax:
    content: protected CryptographicException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);
    parameters:
    - id: info
      type: System.Runtime.Serialization.SerializationInfo
      description: "Объект, содержащий сериализованные данные объекта."
    - id: context
      type: System.Runtime.Serialization.StreamingContext
      description: "Контекстные сведения об источнике или назначении."
  overload: System.Security.Cryptography.CryptographicException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Security.Cryptography.CryptographicException.#ctor(System.String,System.Exception)
  id: '#ctor(System.String,System.Exception)'
  parent: System.Security.Cryptography.CryptographicException
  langs:
  - csharp
  name: CryptographicException(String,Exception)
  nameWithType: CryptographicException.CryptographicException(String,Exception)
  fullName: System.Security.Cryptography.CryptographicException.CryptographicException(String,Exception)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Security.Cryptography
  summary: "Инициализирует новый экземпляр <xref href=&quot;System.Security.Cryptography.CryptographicException&quot;> </xref> класса с указанным сообщением об ошибке и ссылкой на внутреннее исключение, вызвавшее данное исключение."
  remarks: "Исключение, созданное как прямой результат предыдущего исключения, должно содержать ссылку на предыдущее исключение в <xref:System.Exception.InnerException%2A>свойство.</xref:System.Exception.InnerException%2A> <xref:System.Exception.InnerException%2A>Свойство возвращает то же значение, которое передается конструктору, или `null` Если <xref:System.Exception.InnerException%2A>свойства не предоставляет конструктору значение внутреннего исключения.</xref:System.Exception.InnerException%2A> </xref:System.Exception.InnerException%2A>       В следующей таблице показаны исходные значения свойств для экземпляра класса <xref:System.Security.Cryptography.CryptographicException>.</xref:System.Security.Cryptography.CryptographicException>      | Свойство | Значение |   |--------------|-----------|   | <xref:System.Exception.InnerException%2A>| Ссылка на внутреннее исключение. |   | <xref:System.Exception.Message%2A>| Строка сообщения об ошибке. |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A>"
  example:
  - "The following code example demonstrates how to construct a <xref:System.Security.Cryptography.CryptographicException> using a custom error message and an inner exception. This code example is part of a larger example provided for the <xref:System.Security.Cryptography.CryptographicException> class.  \n  \n [!code-cs[System.Security.Cryptography.CryptographicException#5](~/add/codesnippet/csharp/d23f33f8-33a5-45ad-932a-_1.cs)]\n [!code-vb[System.Security.Cryptography.CryptographicException#5](~/add/codesnippet/visualbasic/d23f33f8-33a5-45ad-932a-_1.vb)]\n [!code-cpp[System.Security.Cryptography.CryptographicException#5](~/add/codesnippet/cpp/d23f33f8-33a5-45ad-932a-_1.cpp)]"
  syntax:
    content: public CryptographicException (string message, Exception inner);
    parameters:
    - id: message
      type: System.String
      description: "Сообщение об ошибке с объяснением причин исключения."
    - id: inner
      type: System.Exception
      description: "Исключение, которое стало причиной текущего исключения. Если `inner` не <xref uid=&quot;langword_csharp_null&quot; name=&quot;null&quot; href=&quot;&quot;> </xref>, текущее исключение сгенерировано в <xref uid=&quot;langword_csharp_catch&quot; name=&quot;catch&quot; href=&quot;&quot;> </xref> блок, обрабатывающем внутреннее исключение."
  overload: System.Security.Cryptography.CryptographicException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Security.Cryptography.CryptographicException.#ctor(System.String,System.String)
  id: '#ctor(System.String,System.String)'
  parent: System.Security.Cryptography.CryptographicException
  langs:
  - csharp
  name: CryptographicException(String,String)
  nameWithType: CryptographicException.CryptographicException(String,String)
  fullName: System.Security.Cryptography.CryptographicException.CryptographicException(String,String)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Security.Cryptography
  summary: "Инициализирует новый экземпляр <xref href=&quot;System.Security.Cryptography.CryptographicException&quot;> </xref> класса с указанным сообщением об ошибке в указанном формате."
  remarks: "В следующей таблице показаны исходные значения свойств для экземпляра класса <xref:System.Security.Cryptography.CryptographicException>.</xref:System.Security.Cryptography.CryptographicException>      | Свойство | Значение |   |--------------|-----------|   | <xref:System.Exception.InnerException%2A>| `null`.|   | <xref:System.Exception.Message%2A>| Строка сообщения об ошибке. |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A>"
  example:
  - "The following code example demonstrates how to construct a <xref:System.Security.Cryptography.CryptographicException> using a time format and the current date. This code example is part of a larger example provided for the <xref:System.Security.Cryptography.CryptographicException> class.  \n  \n [!code-cs[System.Security.Cryptography.CryptographicException#6](~/add/codesnippet/csharp/b79fae48-8bdf-49eb-8d86-_1.cs)]\n [!code-vb[System.Security.Cryptography.CryptographicException#6](~/add/codesnippet/visualbasic/b79fae48-8bdf-49eb-8d86-_1.vb)]\n [!code-cpp[System.Security.Cryptography.CryptographicException#6](~/add/codesnippet/cpp/b79fae48-8bdf-49eb-8d86-_1.cpp)]"
  syntax:
    content: public CryptographicException (string format, string insert);
    parameters:
    - id: format
      type: System.String
      description: "Формат, используемый для вывода сообщения об ошибке."
    - id: insert
      type: System.String
      description: "Сообщение об ошибке с объяснением причин исключения."
  overload: System.Security.Cryptography.CryptographicException.#ctor*
  exceptions: []
  platform:
  - net462
references:
- uid: System.SystemException
  isExternal: false
  name: System.SystemException
- uid: System.Security.Cryptography.CryptographicException.#ctor
  parent: System.Security.Cryptography.CryptographicException
  isExternal: false
  name: CryptographicException()
  nameWithType: CryptographicException.CryptographicException()
  fullName: System.Security.Cryptography.CryptographicException.CryptographicException()
- uid: System.Security.Cryptography.CryptographicException.#ctor(System.Int32)
  parent: System.Security.Cryptography.CryptographicException
  isExternal: false
  name: CryptographicException(Int32)
  nameWithType: CryptographicException.CryptographicException(Int32)
  fullName: System.Security.Cryptography.CryptographicException.CryptographicException(Int32)
- uid: System.Int32
  parent: System
  isExternal: true
  name: Int32
  nameWithType: Int32
  fullName: System.Int32
- uid: System.Security.Cryptography.CryptographicException.#ctor(System.String)
  parent: System.Security.Cryptography.CryptographicException
  isExternal: false
  name: CryptographicException(String)
  nameWithType: CryptographicException.CryptographicException(String)
  fullName: System.Security.Cryptography.CryptographicException.CryptographicException(String)
- uid: System.String
  parent: System
  isExternal: true
  name: String
  nameWithType: String
  fullName: System.String
- uid: System.Security.Cryptography.CryptographicException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  parent: System.Security.Cryptography.CryptographicException
  isExternal: false
  name: CryptographicException(SerializationInfo,StreamingContext)
  nameWithType: CryptographicException.CryptographicException(SerializationInfo,StreamingContext)
  fullName: System.Security.Cryptography.CryptographicException.CryptographicException(SerializationInfo,StreamingContext)
- uid: System.Runtime.Serialization.SerializationInfo
  parent: System.Runtime.Serialization
  isExternal: false
  name: SerializationInfo
  nameWithType: SerializationInfo
  fullName: System.Runtime.Serialization.SerializationInfo
- uid: System.Runtime.Serialization.StreamingContext
  parent: System.Runtime.Serialization
  isExternal: true
  name: StreamingContext
  nameWithType: StreamingContext
  fullName: System.Runtime.Serialization.StreamingContext
- uid: System.Security.Cryptography.CryptographicException.#ctor(System.String,System.Exception)
  parent: System.Security.Cryptography.CryptographicException
  isExternal: false
  name: CryptographicException(String,Exception)
  nameWithType: CryptographicException.CryptographicException(String,Exception)
  fullName: System.Security.Cryptography.CryptographicException.CryptographicException(String,Exception)
- uid: System.Exception
  parent: System
  isExternal: true
  name: Exception
  nameWithType: Exception
  fullName: System.Exception
- uid: System.Security.Cryptography.CryptographicException.#ctor(System.String,System.String)
  parent: System.Security.Cryptography.CryptographicException
  isExternal: false
  name: CryptographicException(String,String)
  nameWithType: CryptographicException.CryptographicException(String,String)
  fullName: System.Security.Cryptography.CryptographicException.CryptographicException(String,String)
- uid: System.Security.Cryptography.CryptographicException.#ctor*
  parent: System.Security.Cryptography.CryptographicException
  isExternal: false
  name: CryptographicException
  nameWithType: CryptographicException.CryptographicException
